### ISSUE TITLE
Redirecting stdout to stderr doesn't work

### ISSUE THREAD

**Issue by kseistrup** [xonsh/xonsh#323]
In bash I usually write error messages to `stderr`, e.g.:

``` sh
[ CONDITION_THAT_SHOULD_SUCCEED ] || {
  echo 'we failed' >&2
  exit 1
}
echo 'we succeeded'
```

Following [the tutorial](http://xonsh.org/tutorial.html#input-output-redirection), I would expect at least one of these to work:

``` sh
echo 'we failed' out>err
echo 'we failed' >&2        # included for bash compatibility
```

but none of them do.

**Comment by scopatz** [comment]
Hi @kseistrup - this looks like a legitimate bug.  We'll try to get to it quickly.  Alternatively, I can tell you where to start looking, if you want to lend a hand with debugging.

**Comment by kseistrup** [comment]
I know where to look (and I did look).  The reason I didn't try to fix it is because I don't understand how the lexer works and what to do about it. ;)

**Comment by scopatz** [comment]
Ahh OK.  I'll take a look tonight.  That is all @adqm's work :)

**Comment by adqm** [comment]
I don't know that I would quite qualify this as a bug; this was intentionally left out as a feature.  It isn't by any means impossible to implement, but it does require some effort.  I tried to make this clear in the tutorial, but, evidently, I failed.

The constructor for `subprocess.Popen` objects accepts `stderr=STDOUT` to make stderr go to stdout, but it doesn't offer the opposite.  We will have to implement that ourselves if we want it (this will, I think, make the lexer less complicated, but it will make the handling of those subprocesses harder, as we'll have to monkey with the pipes ourselves).

**Comment by scopatz** [comment]
Ahh I see.  It should still be possible to set `stdout=sys.stderr`, though.  But you are right, and I missed this, that we'll have to do something different.

**Comment by kseistrup** [comment]
@adqm, I'm sorry if I didn't read the docs thoroughly enough, but as @scopatz said: it should still be possible to set `stdout=sys.stderr` in the `Popen()` call.

**Comment by adqm** [comment]
Is `stdout=sys.stderr` actually what we want?  It's not clear to me that that will interact with other redirects properly (that is, if we did `o>e` and also sent stderr or stdout to a file, would we still get the right thing?). 

We can always try it, but I think we may need to dig in a little deeper (in the `_get_handles` method of the `Popen` and `ProcProxy` classes, where `stderr=STDOUT` is handled).

**Comment by scopatz** [comment]
I am not sure that it is exactly what we want in the long term, but it might be good enough for the short to medium term.  It is worth digging into a bit...

**Comment by scopatz** [comment]
A potential fix for this is in #2081
